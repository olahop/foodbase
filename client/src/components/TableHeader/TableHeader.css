Th {
    font-weight: 1200;
    color: white;
    text-align: left;
    border-bottom-style: solid;
    border-bottom-width: 1px;
    border-bottom-color: white;
    padding: 10px 0 10px 0;
    width: 25%;
    display: block;
    cursor: pointer;
}

Thead Tr{
    width: 100%;
    display: flex;
}

Thead{
    display: flex;
}
