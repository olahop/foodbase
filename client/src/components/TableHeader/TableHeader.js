import React from "react";
import { connect } from "react-redux";
import { changeSort } from "./../../actions/sort";
import { resetSortAsc, toggleSortAsc } from "./../../actions/sortAsc";
import { Thead, Tr, Th } from "react-super-responsive-table";
import "./TableHeader.css";
import "react-super-responsive-table/dist/SuperResponsiveTableStyle.css";


class TableHeader extends React.Component {

    onChange = event => {
        if (this.props.sort !== event.target.getAttribute("value")){
            this.props.onChangeSort(event.target.getAttribute("value"));
            this.props.onChangeResetAsc();
        }else{
            this.props.onChangeToggleAsc();
        }
    }

    render() {
        const { sort, sortAsc } = this.props
        return (
            <Thead>
                <Tr>
                    <Th className="header-option" value="Name" onClick={this.onChange}>
                        Name {sort === "Name" ? (sortAsc ? "↓" : "↑") : ""}
                    </Th>
                    <Th className="header-option" value="Fat" onClick={this.onChange}>
                        Fat&nbsp;(g) {sort === "Fat" ? (sortAsc ? "↓" : "↑") : ""}
                    </Th>
                    <Th className="header-option" value="Carbohydrates" onClick={this.onChange}>
                        Carbs&nbsp;(g) {sort === "Carbohydrates" ? (sortAsc ? "↓" : "↑") : ""}
                    </Th>
                    <Th className="header-option" value="Protein" onClick={this.onChange}>
                        Protein&nbsp;(g) {sort === "Protein" ? (sortAsc ? "↓" : "↑") : ""}
                    </Th>
                </Tr>
            </Thead>
        );
    }
}

const mapStateToProps = state => ({
    sort: state.sort,
    sortAsc: state.sortAsc
});

const mapActionsToProps = {
    onChangeSort: changeSort,
    onChangeToggleAsc: toggleSortAsc,
    onChangeResetAsc: resetSortAsc
};

export default connect(mapStateToProps,mapActionsToProps)(TableHeader)
