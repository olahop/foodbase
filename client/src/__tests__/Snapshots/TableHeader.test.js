import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import TableHeader from '../../components/TableHeader/TableHeader';

const mockStore = configureStore([]);
describe('Test App component', () => {
  let store;
  let component;
  
  beforeAll(() => {
    store = mockStore();
    component = renderer.create(
      <Provider store={store}>
        <TableHeader />
      </Provider>
    );
  });

  it('Component renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
//TODO: add triggerSort-test
/*it('test triggerSort', async () => {

});*/
