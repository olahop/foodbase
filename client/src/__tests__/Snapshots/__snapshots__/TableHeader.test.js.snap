// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Component renders correctly 1`] = `
<thead>
  <tr>
    <th
      className="header-option"
      onClick={[Function]}
      value="Name"
    >
      Name 
      
    </th>
    <th
      className="header-option"
      onClick={[Function]}
      value="Fat"
    >
      Fat (g) 
      
    </th>
    <th
      className="header-option"
      onClick={[Function]}
      value="Carbohydrates"
    >
      Carbs (g) 
      
    </th>
    <th
      className="header-option"
      onClick={[Function]}
      value="Protein"
    >
      Protein (g) 
      
    </th>
  </tr>
</thead>
`;
